mindmeld/components/role_classifier.py

```diff
@@ -127,16 +125,21 @@ def fit(self,
             model.fit(examples, labels)
             self._model = model
             self.config = ClassifierConfig.from_model_config(self._model.config)
+        else:
+            # This _else_ conditional is created to support moving .dump() and .load()
+            # to self._model.dump() and self._model.load() respectively.
+            #
+            # Any calls to abstract methods other than .fit(), .load(), .unload(), and .dump() are
+            # to be carefully used by identifying if role classification is valid or not with the
+            # conditional`if len(self.role_classifier.roles) > 1: ...; else: ...`
+            self._model = model
```

Comment:
If we have no valid examples to train on, shouldn't `self._model` be set to `None`?

Reply:
tldr: Sure, will make the required change.

On top of the comment in previous lines, I realized we can do `self._model=None` and put a try-catch in the load method wherein if there exists no model, the load_model() (based on AutoModel) throws an error and upon catching the error, the model is set to None in load method as well.